depcuiser-metrics-report/action.yml

```
@@ -0,0 +1,39 @@
+name: 'Dependency cruiser metrics report'
+description: 'Generates a report for the project of dependency cruiser metrics'
+
+inputs:
+  npm-token:
+    description: Access token type **Read-only**. Required for repository with private dependencies. If undefined, `env.NPM_TOKEN` is used
+    required: false
+  report-output-path:
+    description: Path to the metrics report json file
+    default: dependency-cruiser-results.json
+    required: false
+  tsconfig-path:
+    description: Path to the custom tsconifg.json file
+    default: ./tsconfig.json
+    required: false
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install dependency cruiser
+      shell: bash
+      env:
+        NPM_TOKEN: ${{ inputs.npm-token || env.NPM_TOKEN }}
+      run: yarn add --dev dependency-cruiser -W
+
+    - name: Add tsconfig to include all available files
+      shell: bash
+      run: |
+        echo "{
+          \"extends\": \"${{ inputs.tsconfig-path }}\",
+          \"include\": [\"**/*\"]
+        }" > tsconfig.dependency-cruiser.json
```

When using inputs in shell scripts its safer to convert it to env first to prevent injections 🙏 
```suggestion
      env:
        TS_CONFIG: ${{ inputs.tsconfig-path }}
      run: |
        echo "{
          \"extends\": \"${{ env.TS_CONFIG }}\",
          \"include\": [\"**/*\"]
        }" > tsconfig.dependency-cruiser.json
```

---

@OndrejTuma fixed 👍 thank you!